src/utils/workflow-utils.ts

```diff
@@ -242,7 +258,18 @@ export const generateTaskSpec = (
                         resource: {
                             ...resourceConfig,
                         },
-                    });
+                    };
+
+                    if (
+                        entityType === 'materialization' &&
+                        incompatibleSchemaChanges &&
+                        hasLength(incompatibleSchemaChanges[bindingUUID])
+                    ) {
```

Comment:
If we're duplicating this check this it should be stored off so they always stay in sync.

I don't think we van get around checking it so putting it up with `bindingDisabled` and `existingBindingIndex` probably makes sense.